### PR TITLE
Options API

### PR DESCRIPTION
**Pull request by jsteemann**
### Scope & Purpose

Options API:

* Expose REST API endpoints for server options discovery:
  - GET `/_admin/options-description`: returns a JSON description of all available server options.
  - GET `/_admin/options`: returns a JSON object with the currently set server options.

  As these API may reveal sensitive data about the deployment, they can only
  be accessed from inside the `_system` database. In addition, there is a
  policy control startup option `--server.options-api` that determines if and
  to whom the APIs are made available. This option can have the following
  values:
  - `disabled`: options APIs is disabled.
  - `jwt`: options APIs can only be accessed via superuser JWT.
  - `admin`: options APIs can be accessed by admin users in the `_system` database only.
  - `public`: everyone with access to `_system` database can access the options APIs.

- [ ] :hankey: Bugfix
- [x] :pizza: New feature
- [ ] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification

### Checklist

- [ ] Tests
  - [ ] **Regression tests**
  - [ ] C++ **Unit tests**
  - [ ] **integration tests**
  - [ ] **resilience tests**
- [x] :book: CHANGELOG entry made
- [ ] :books: documentation written (release notes, API changes, ...)
- [ ] Backports
  - [ ] Backport for 3.11: -
  - [ ] Backport for 3.10: -

#### Related Information

- [ ] Docs PR: 
- [ ] Enterprise PR:
- [ ] GitHub issue / Jira ticket:
- [ ] Design document:

### CODE REVIEW

File: arangod/RestHandler/RestOptionsBaseHandler.h [hunk 1]
```diff
@@ -0,0 +1,39 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2014-2023 ArangoDB GmbH, Cologne, Germany
```
Comment (joerg84) [inline arangod/RestHandler/RestOptionsBaseHandler.h:4]: 2014-2024 :-)

File: arangod/RestHandler/RestOptionsHandler.cpp [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    return RestStatus::DONE;
+  }
+
+  if (!checkAuthentication()) {
+    // checkAuthentication() will have created the response message already
+    return RestStatus::DONE;
+  }
+
+  // filter out these attributes from the options
+  auto filter = [](std::string const& name) {
```
Comment (joerg84) [inline arangod/RestHandler/RestOptionsHandler.cpp:53]: How can we make sure/document new sensitive options will be added here as well?

File: arangod/RestHandler/RestOptionsBaseHandler.h (hunk 1 above)
Comment (jsteemann) [reply arangod/RestHandler/RestOptionsBaseHandler.h:4]: You're right. I guess we should address this separately and "fix" all headers at the same time with a script.

File: arangod/RestHandler/RestOptionsHandler.cpp (hunk 2 above)
Comment (jsteemann) [reply arangod/RestHandler/RestOptionsHandler.cpp]: Moved this into a central function now.

**Review by neunhoef — APPROVED** [review]
LGTM

**Pull request merged**